boards/esp32: adds Heltec WiFi LoRa 32 V2 board definition

gschorcht opened the pull request:
### Contribution description

This PR adds the board definition for Heltec WiFi LoRa 32 V2 board. Heltec WiFi LoRa 32 V2 is an ESP32 development board with 8 MB Flash that uses the EPS32 chip directly. It integrates 

- a SemTech SX1276 or SX1278 for LoRaWAN communication and
- a SSD1306 128x64 OLED display connected via I2C.

Many GPIOs are broken out. Since the board is [open source hardware](https://github.com/Heltec-Aaron-Lee/WiFi_Kit_series), a number of clones are available.

### Testing procedure

Application `examples/lorawan` can be used to test the board including the LoRaWAN functionality.
```
DRIVER=sx127x make BOARD=esp32-heltec-lora32-v2 -C examples/lorawan/ flash
```

### Issues/PRs references

Depends on PR #13685

aabadie commented on boards/esp32-heltec-lora32-v2/Makefile.dep:
```diff
@@ -0,0 +1 @@
+include $(RIOTBOARD)/common/esp32/Makefile.dep
```
Here you could also add (maybe it has other implication with esp stuff ?):
```mk
ifneq (,$(filter netdev_default,$(USEMODULE)))
  USEMODULE += sx1276
endif
```

gschorcht replied:
Hm, ..., not sure what the most usual case is. Default network interface might be also `esp_now` or `esp_wifi`.

benpicco replied:
Why is it either or?
With this you'd get both `sx1276` *and* `esp_now`.

But this can be added when #12994 is merged so the user doesn't have to be careful not to forget to adding `GNRC_NETIF_NUMOF=2`, or else run into an assert failure when the pre-allocated netif slots run out.

yegorich commented on boards/esp32-heltec-lora32-v2/doc.txt:
```diff
@@ -0,0 +1,210 @@
+Heltec WiFi LoRa 32 V2 is an ESP32 development board with 8 MB Flash that uses
+the EPS32 chip directly. It integrates 
+
+- a SemTech SX1276 or SX1276 for LoRaWAN communication and
```
do you mean SX1278?

gschorcht replied:
Yes of course :worried:

gschorcht commented:
@yegorich Squased.

yegorich commented on boards/esp32-heltec-lora32-v2/doc.txt:
```diff
@@ -0,0 +1,210 @@
+@note
+- The configuration of ADC channels contains all ESP32 GPIOs that can be used
+  as ADC channels.
+- GPIO9 and GIOP10 can only be used in **dout** and **dio**
```
s/GIOP10/GPIO10

gschorcht replied:
Done

MrKevinWeiss commented:
release status ping, should I change the milestone?

gschorcht commented:
@MrKevinWeiss Sorry for late answer, I'm quite busy at the moment. From my point of view, it is ready for release. I'm just waiting for a trigger to squash it.

gschorcht commented:
@MrKevinWeiss Sorry for late answer, I'm quite busy at the moment. From my point of view, it is ready for release. I'm just waiting for a trigger to squash it.

MrKevinWeiss commented:
Does anyone else have this board?  If not I can add it to the wishlist but then I don't think it will get in to this release.  Please advise otherwise I will move the milestone.

MrKevinWeiss commented:
I think we can wait until next release for this one.

fjmolinas commented:
@gschorcht if you can run `compile_and_test_for_board` and post the output it would be enough to move forward with this one.

yegorich commented on boards/esp32-heltec-lora32-v2/doc.txt:
```diff
@@ -0,0 +1,210 @@
+
+- SemTech SX1276 for LoRaWAN communication in the 433 MHz band
+- SemTech SX1278 for LoRaWAN communication in the 868/915 MHz band
+
```
Those two chips should be swapped as it was made in https://github.com/RIOT-OS/RIOT/pull/13389.

gschorcht replied:
Done

yegorich commented:
@gschorcht I think, the only thing missing is `board_init()` in `board.h`.

gschorcht commented:
> @gschorcht I think, the only thing missing is `board_init()` in `board.h`.

Yes, it will become a problem once I rebased it.

gschorcht commented:
With the last commit, I added a small section on how to use the board's display.

gschorcht commented:
May I squash this PR?

benpicco commented:
More than a month has passed, I'd say it's safe to squash :wink:

gschorcht commented:
Squashed. But I have removed the `USEMODULE += boards_common_esp32` from `Makefile.dep`. It will be pulled in by `boards/common/esp32/Makefile.dep` once PR #13685 is merged. We should wait for this cleanup.

Another question remains. @aabadie suggested to use `sx1276` as netdev default.
```c
ifneq (,$(filter netdev_default,$(USEMODULE)))
  USEMODULE += sx1276
endif
```
This would mean that it would pull in `esp_now` and `sx1276`. Would this be OK?

benpicco commented on boards/esp32-heltec-lora32-v2/include/periph_conf.h:
```diff
@@ -0,0 +1,179 @@
+/**
+ * @name   UART configuration
+ *
+ * ESP32 provides 3 UART interaces at maximum:
```
```suggestion
 * ESP32 provides 3 UART interfaces at maximum:
```

benpicco commented:
You can squash again :wink:

gschorcht commented:
@benpico Thanks for reviewing and merging.
